aws/resource_aws_greengrass_resource_definition.go

```diff
@@ -0,0 +1,627 @@
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"auto_add_group_owner": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  false,
+						},
+						"group_owner": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func generateLocalVolumeResourceDataSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"source_path": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"destination_path": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"group_owner_setting": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"auto_add_group_owner": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  false,
+						},
+						"group_owner": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func generateS3MachineLearningModelResourceDataSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"s3_uri": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"destination_path": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func generateSageMakerMachineLearningModelResourceDataSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"sagemaker_job_arn": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateArn,
+			},
+			"destination_path": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func generateSecretsManagerSecretResourceDataSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"secret_arn": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateArn,
+			},
+			"additional_staging_labels_to_download": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func resourceAwsGreengrassResourceDefinition() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsGreengrassResourceDefinitionCreate,
+		Read:   resourceAwsGreengrassResourceDefinitionRead,
+		Update: resourceAwsGreengrassResourceDefinitionUpdate,
+		Delete: resourceAwsGreengrassResourceDefinitionDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"amzn_client_token": {
+				Type:      schema.TypeString,
+				Optional:  true,
+				Sensitive: true,
+			},
+			"tags": tagsSchema(),
+			"latest_definition_version_arn": {
```

Comment:
It might be good to add `last_updated_timestamp` and `latest_version` `Computed` attributes here. They should also be flagged by the `CustomizeDiff` function